### PR TITLE
support for simulator recorder

### PR DESCRIPTION
**Pull request by pelikhan**
Adds a "recorder" button that allows to record animated gifs of the simulator.
TODO:
- [ ] 3rd party notice
- [ ] add logo
- [ ] support for FF
- [ ] trap all UI updates

recorder

- simulator sends XML snapshots after updating state
- rendered in UI thread
- assembed into gif using gifjs

### CODE REVIEW

**Comment by abchatra** [comment]
Is this tested on IE11 & Edge & mobile browsers?

File: webapp/public/gifjs/gif.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// gif.js 0.2.0 - https://github.com/jnordberg/gif.js
```
Comment (abchatra) [inline webapp/public/gifjs/gif.js]: > https://github.com/jnordberg [](start = 17, length = 29)

Is this approved by open source request?

Comment (pelikhan) [inline webapp/public/gifjs/gif.js]: Yes

**Comment by pelikhan** [comment]
@microsoftsam the simulator CSS does not seem to be complete anymore. Do we have dependencies on sim.css?

**Comment by jaustin** [comment]
We got asked about this feature again recently, is it 'coming back'?

**Comment by abchatra** [comment]
We already have various way to share which are under utilized. 
1. Download screenshot
2. Download Hex
3. Share the link
4. Embed
I am not sure adding to the list will help the broader set of audience. We are still evaluating. This is unlikely to make it to V0.

**Pull request closed without merging**